### PR TITLE
Add support for "when timer greater than"

### PR DESCRIPTION
**Pull request by adroitwhiz**
Resolves part of #122.

This PR adds support for "edge-activated" triggers, so named because they evaluate a predicate every step (e.g. "is the timer greater than N?") and trigger their script on the rising edge (when the predicate changes from `false` to `true`).

Trigger predicates/options can now be functions (which are passed the current target) as well as values. This is necessary because the input to a "greater than" block is actually droppable and can contain arbitrary scripts. For codegen purposes, it may be easier to have the function take *no* arguments and just bind the `this` value to the current target, but that seems jankier.

### CODE REVIEW

File: src/Project.js [hunk 1]
```diff
@@ -129,19 +176,9 @@ export default class Project {
       }
     }
 
-    // Find triggers which match conditions
-    let matchingTriggers = [];
-    for (let i = 0; i < this.spritesAndStage.length; i++) {
-      const sprite = this.spritesAndStage[i];
```
Comment (adroitwhiz) [inline src/Project.js:135]: Interesting quirk about the old code I noticed: `spritesAndStage` is actually a getter which returns a new array every time it's called 😬

**Comment by PullJosh** [comment]
Right now it looks like the trigger is timer-specific. Could it be better to simply have a `WHEN_TRUE` trigger that takes a function as input and activates on the rising edge? (Might as well add `WHEN_FALSE` as well.)

That way you could trigger when loudness is greater than some number, but also when other conditions are true.

**Comment by adroitwhiz** [comment]
I thought about that--I was kinda worried there's some reason Scratch itself hasn't just implemented that.

It seems that Scratch previously [wanted to implement a "when touching" block but ran into some issues](https://github.com/LLK/scratch-vm/issues/1532)--for instance, [a script running in parallel that messes with the predicate you're testing for](https://github.com/LLK/scratch-gui/issues/2300), which *technically* makes logical sense but is very unintuitive behavior.

**Comment by adroitwhiz** [comment]
In general, I think any sort of "when predicate" block is unintuitive because you have to be acutely aware of execution order (IIRC, there were a ton of bugs in 3.0 relating to edge cases in the "when timer" block, showing that the developers themselves didn't even understand the behavior).

I'd prefer to steer users away from using a "when this predicate evaluates to true" block and only implementing the hat blocks that are necessary to ensure compatibility with Scratch.

**Comment by PullJosh** [comment]
Okay, that makes sense. I am on board with the current choice then. 👍

File: src/Project.js (hunk 1 above)
Comment (PullJosh) [reply src/Project.js:135]: [Who](https://github.com/leopard-js/leopard/blame/1e4e89edbb708c2e12f472c47d1c17f552d36152/src/Project.js#L177) would ever write such a thing 🙄

File: src/Project.js [hunk 2]
```diff
@@ -81,7 +84,51 @@ export default class Project {
...
+
+  _stepEdgeActivatedTriggers() {
+    const edgeActivated = this._matchingTriggers(tr => tr.isEdgeActivated);
+    const triggersToStart = [];
+    for (const triggerWithTarget of edgeActivated) {
+      const { trigger } = triggerWithTarget;
+      let predicate;
+      switch (trigger.trigger) {
+        case Trigger.TIMER_GREATER_THAN:
+          predicate = this.timer > trigger.option("VALUE");
```
Comment (PullJosh) [inline src/Project.js:110]: You added support for function options which can receive the `target` as an argument, but then you are not passing `target` into `trigger.option()` here. Is there a reason why you don't pass it?

Comment (adroitwhiz) [reply src/Project.js]: Nope, just an oversight

**Review by PullJosh — APPROVED** [review]
Behavior now looks correct to me :)

**Pull request merged**